tests/pass/no_std.rs

```diff
@@ -0,0 +1,18 @@
+#![feature(lang_items, start)]
+#![no_std]
+// rustc-env: MIRI_NO_STD=1
```

Comment:
This doesn't really help since the env var is for cargo-miri, isn't it?

Reply:
This is needed to make the windows test suite pass. The env var is also used by the tls dtor avoidance logic

Reply:
Yeah we should either make that just silently do nothing when the symbol is missing, or actually scan the link sections.

Not running the test also works for me.

Reply:
Silently doing nothing causes the memory leak checker to complain 😆 it was the first thing I tried

Reply:
I mean doing nothing *if there is no std*. That should not cause any leaks since you cannot have any thread-locals.